README.md

```diff
@@ -214,14 +199,88 @@ without the JNI the performance can suffer.
 A quick way to check if a JAR contains such shared objects is to simply unzip it and check if
 any of the resulting files are shared-objects and if an aarch64 (arm64) shared-object is missing:
 ```
-$ unzip foo.jar
+$ jar xf foo.jar
 $ find . -name "*.so" | xargs file
 ```
 For each x86-64 ELF file, check there is a corresponding aarch64 ELF file
-in the binaries. With some common packages (e.g. commons-crypto) we've seen that
+in the binaries. With some common packages \(e.g. commons-crypto\) we've seen that
 even though a JAR can be built supporting Arm manually, artifact repositories such as
 Maven don't have updated versions. Feel free to open an issue in this GitHub repo for advice.
 
+### Building multi-arch JARs
+Java is meant to be a write once, and run anywhere language.  When building Java artifacts that
+contain native code, it is important to build those libraries for each major architecture to provide
+a seamless and optimally performing experience for all consumers.  Code that runs well on both Graviton and x86
+based instances increases the package's utility.
+
+There is nominally a multi-step process to build the native shared objects for each supported
+architecture before doing the final packaging with Maven, SBT, Gradle etc. Below is an example
+of how to create your JAR using Maven that contains shared libraries for multiple distributions
+and architectures for running your Java application interchangeably on AWS EC2 instances
+based on x86 and Graviton processors:
+
+```
+# Create two build instances, one x86 and one Graviton instance.
+# Pick on instance to be the 'master' instance.
+# Log into the non-master instance
+$ cd java-lib
+$ mvn package
+$ find target/ -name "*.so" -type f -print
+
+# Note the directory this so file is in, it will be in a directory
+# such as: target/classes/org/your/class/hierarchy/native/OS/ARCH/lib.so
+
+# Log into the master build instance
+$ cd java-lib
+$ mvn package
+
+# Repeat the below two steps for each OS and ARCH combination you want to release
+$ mkdir target/classes/org/your/class/hierarchy/native/OS/ARCH
+$ scp slave:~/your-java-lib/target/classes/org/your/class/hierarchy/native/OS/ARCH/lib.so target/classes/org/your/class/hierarchy/native/OS/ARCH/
+
+# Do final jar packaging with maven.  It will include the additional
+# native libraries even though they were not built directly by this maven process.
+$ mvn package
+
+# To do a release to Maven Central and/or Sonatype Nexus:
+$ mvn release:prepare
+$ mvn release:perform
+
+# When creating a single Jar for all platform native libraries, 
+# the release plugin's configuration must be modified to specify 
+# the plugin's `preparationGoals` to not include the clean goal.
+# See http://maven.apache.org/maven-release/maven-release-plugin/prepare-mojo.html#preparationGoals
```

Comment:
This seems to come pretty late... I imagine someone would execute all the steps and then be like ohh.. it didn't work.

Reply:
Moved this up.